homeassistant/components/flexit_bacnet/climate.py

```diff
@@ -0,0 +1,174 @@
+"""The Flexit Nordic (BACnet) integration."""
+import asyncio.exceptions
+from typing import Any
+
+from flexit_bacnet import (
+    VENTILATION_MODE_AWAY,
+    VENTILATION_MODE_HOME,
+    VENTILATION_MODE_STOP,
+    FlexitBACnet,
+)
+from flexit_bacnet.bacnet import DecodingError
+
+from homeassistant.components.climate import (
+    PRESET_AWAY,
+    PRESET_BOOST,
+    PRESET_HOME,
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    DOMAIN,
+    PRESET_TO_VENTILATION_MODE_MAP,
+    VENTILATION_TO_PRESET_MODE_MAP,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_devices: AddEntitiesCallback,
+) -> None:
+    """Set up the Flexit Nordic unit."""
+    device = hass.data[DOMAIN][config_entry.entry_id]
+
+    async_add_devices([FlexitClimateEntity(device)])
+
+
+class FlexitClimateEntity(ClimateEntity):
+    """Flexit air handling unit."""
+
+    _attr_has_entity_name = True
+
+    _attr_hvac_modes = [
+        HVACMode.OFF,
+        HVACMode.FAN_ONLY,
+    ]
+
+    _attr_preset_modes = [
+        PRESET_AWAY,
+        PRESET_HOME,
+        PRESET_BOOST,
+    ]
+
+    _attr_supported_features = (
+        ClimateEntityFeature.PRESET_MODE
+        | ClimateEntityFeature.TARGET_TEMPERATURE
+        | ClimateEntityFeature.AUX_HEAT
+    )
+
+    _attr_target_temperature_step = 1.0
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+
+    def __init__(self, device: FlexitBACnet) -> None:
+        """Initialize the unit."""
+        self._device = device
+        self._attr_unique_id = device.serial_number
+        self._attr_device_info = DeviceInfo(
+            identifiers={
+                (DOMAIN, device.serial_number),
+            },
+            name="Ventilation",
```

Comment:
Why don't we set the device name here? `self._device.device_name`

Reply:
Device name would then be "HvacFnct21y_A" (for my unit). Or should we try to get the model here instead. I.e. in my case that would be "Flexit Nordic S4".

Reply:
As long as the user can identify it